tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/internal/UpgradePlannerService.java

```diff
@@ -204,6 +217,65 @@ public void removeListener(UpgradeListener upgradeListener) {
 		_upgradeListeners.remove(upgradeListener);
 	}
 
+	@Override
+	public void removeUpgradePlan(UpgradePlan upgradePlan) {
```

Comment:
we should put the logic of removing xml nodes into XMLMemeto class rather than here directly.